hal/src/nRF52840/ble_hal.cpp

```diff
@@ -15,739 +15,3147 @@
+
+#define CHECK_RETURN(ret, str) \
+        do { \
+            const int _ret = ret; \
+            const char* _str = str; \
+            if (_ret != SYSTEM_ERROR_NONE) { \
+                if (_str != nullptr) { \
+                    LOG(ERROR, "%s failed: %u", _str, _ret); \
+                } \
+                return _ret; \
+            } \
+        } while (false)
 
-LOG_SOURCE_CATEGORY("hal.ble")
 
+#define CHECK_RETURN_VOID(ret, str) \
+        do { \
+            const int _ret = ret; \
+            const char* _str = str; \
+            if (_ret != SYSTEM_ERROR_NONE) { \
+                if (_str != nullptr) { \
+                    LOG(ERROR, "%s failed: %u", _str, _ret); \
+                } \
+                return; \
+            } \
+        } while (false)
+
+
+StaticRecursiveMutex s_bleMutex;
+class bleLock {
+    static void lock() {
+        s_bleMutex.lock();
+    }
+    static void unlock() {
+        s_bleMutex.unlock();
+    }
+};
+
+//anonymous namespace
 namespace {
 
-// Advertising interval in 0.625 ms units
-const auto FAST_ADVERT_INTERVAL = MSEC_TO_UNITS(40, UNIT_0_625_MS); // Fast mode
-const auto SLOW_ADVERT_INTERVAL = MSEC_TO_UNITS(500, UNIT_0_625_MS); // Slow mode
+system_error_t sysError(uint32_t error) {
+    switch (error) {
+        case NRF_SUCCESS:                       return SYSTEM_ERROR_NONE;
+        case NRF_ERROR_SVC_HANDLER_MISSING:
+        case NRF_ERROR_NOT_SUPPORTED:           return SYSTEM_ERROR_NOT_SUPPORTED;
+        case NRF_ERROR_SOFTDEVICE_NOT_ENABLED:
+        case NRF_ERROR_INVALID_STATE:           return SYSTEM_ERROR_INVALID_STATE;
+        case NRF_ERROR_INTERNAL:                return SYSTEM_ERROR_INTERNAL;
+        case NRF_ERROR_NO_MEM:                  return SYSTEM_ERROR_NO_MEMORY;
+        case NRF_ERROR_NOT_FOUND:               return SYSTEM_ERROR_NOT_FOUND;
+        case NRF_ERROR_INVALID_PARAM:
+        case NRF_ERROR_INVALID_LENGTH:
+        case NRF_ERROR_INVALID_FLAGS:
+        case NRF_ERROR_INVALID_DATA:
+        case NRF_ERROR_DATA_SIZE:
+        case NRF_ERROR_NULL:                    return SYSTEM_ERROR_INVALID_ARGUMENT;
+        case NRF_ERROR_TIMEOUT:                 return SYSTEM_ERROR_TIMEOUT;
+        case NRF_ERROR_FORBIDDEN:
+        case NRF_ERROR_INVALID_ADDR:            return SYSTEM_ERROR_NOT_ALLOWED;
+        case NRF_ERROR_BUSY:                    return SYSTEM_ERROR_BUSY;
+        case NRF_ERROR_CONN_COUNT:              return SYSTEM_ERROR_LIMIT_EXCEEDED;
+        case NRF_ERROR_RESOURCES:               return SYSTEM_ERROR_ABORTED;
+        default:                                return SYSTEM_ERROR_UNKNOWN;
+    }
+}
+
+bool addressEqual(const hal_ble_addr_t& srcAddr, const hal_ble_addr_t& destAddr) {
+    return (srcAddr.addr_type == destAddr.addr_type && !memcmp(srcAddr.addr, destAddr.addr, BLE_SIG_ADDR_LEN));
+}
+
+} //anonymous namespace
+
+class GattBase {
+public:
+    GattBase() {
+        desiredAttMtu_ = BLE_MAX_ATT_MTU_SIZE;
```

Comment:
Prefer initialization in the constructor member initialization list.

Reply:
Static member can not be initialized in the  constructor member initialization list.

Reply:
In that case prefer initialization at the declaration.